### ISSUE TITLE
[Feature] Limit FPS when in background

### ISSUE THREAD

**Issue by pongo1231** [flightlessmango/MangoHud#754]
Just like the "Background Application Max Frame Rate" option the NVIDIA control panel on Windows provides, this would set a custom FPS limit if a game is out of focus. It's one feature I greatly miss on Linux and would help with keeping the resource usage of games down when there's no reason to still run them at full speed.

**Comment by foundObjects** [comment]
I'd like to chime in on this request too, it would be super handy for multi-boxing.

**Comment by koloved** [comment]
there is actually the **_-o_** option in gamescope, but for me it is not working - 
https://github.com/ValveSoftware/gamescope

**Comment by filen1** [comment]
Here is a script to toggle for anyone else who would like it:
```
#!/bin/bash

CONFIG_FILE="$HOME/.config/MangoHud/MangoHud.conf"
FPS_HIGH=144
FPS_LOW=10

if grep -q "fps_limit=$FPS_HIGH" "$CONFIG_FILE"; then
    sed -i "s/fps_limit=$FPS_HIGH/fps_limit=$FPS_LOW/" "$CONFIG_FILE"
    notify-send -t 2000 "MangoHud" "FPS limit set to $FPS_LOW"
else
    sed -i "s/fps_limit=$FPS_LOW/fps_limit=$FPS_HIGH/" "$CONFIG_FILE"
    notify-send -t 2000 "MangoHud" "FPS limit set to $FPS_HIGH"
fi
```

Create .config/MangoHud/MangoHud.conf and write fps_limit=144 to MangoHud.conf and save.  For ease of use create a panel and add a launcher for the script or assign it to a shortcut. Make sure to have `mangohud %command%` as launch property in steam. This feature would be perfect for games where you want to always be online but afk/doing something else, and the game doesn't natively have it.

**Comment by flightlessmango** [comment]
You don't need a script to toggle fps limits, this is already a feature in mangohud

**Comment by koloved** [comment]
its not working in wayland when i tried this last time

@flightlessmango how can i do this ? let me know please

**Comment by flightlessmango** [comment]
@koloved 
> its not working in wayland when i tried this last time

We've recently added wayland keybinds, so that's probably why it didn't work

**Comment by koloved** [comment]
@flightlessmango yesterday its working for me in intel iGpu + nvidia wayland offload on the fresh install, drop FPS to 30 when alt tabbing, if i will use just nvidia gpu as a main, it will not work
is it only nvidia problem?

**Comment by flightlessmango** [comment]
@koloved I think there's some confusion here. mangohud is not able to change the fps limit based on if it's focused yet. What was discussed is a workaround where you use a keybind to manually toggle the fps limit

**Comment by koloved** [comment]
@flightlessmango seems like it offload feature then, idk 

https://github.com/flightlessmango/MangoHud/assets/22406105/8e77ba88-b970-47d5-bd51-04fb97dcb889

**Comment by flightlessmango** [comment]
I don't think this is feasible to implement in either x11 or wayland

**Comment by autumn-puffin** [comment]
> I don't think this is feasible to implement in either x11 or wayland

I hate to necro this but, at least on the wayland, would this not be possible by setting the fps limit on `wl_keyboard::leave`/`enter` events?

**Comment by flightlessmango** [comment]
Yes I believe it's possible in wayland. I've laid the groundwork in mangohud-next for detecting the focused window, we check for keyboard, pointer and presentation. https://github.com/flightlessmango/MangoHud/blob/73931de948402caf3ea80f26d5bbb6b3f1d2c1d9/mangohud-next/client/wayland.cpp#L256
